Parsing info to notifications screen

issytuttle opened the pull request:
Parsing info to notifications screen

Commit:
Changing typography.
Changes to train line to parse information to Lillian's screen

lilianlyx commented on app/src/main/java/com/example/ptvproject/ui/selecttrainline/SelectTrainLineScreen.kt:
```diff
@@ -1,39 +1,34 @@
     selectTrainLineViewModel: SelectTrainLineViewModel = viewModel(),
-    //onTrainLineSelected: (departureTime: ZonedDateTime, stationName: String, trainLine: String, direction: String) -> Unit
-) {
+    onTrainLineSelected: (String, String, String, String) -> Unit,
```
it would be great to see the parameter names for each of these types, like the older code showed!

lilianlyx commented on app/src/main/java/com/example/ptvproject/ui/selecttrainline/SelectTrainLineScreen.kt:
```diff
@@ -46,16 +41,12 @@ fun SelectTrainLineScreen(
             updateTrainLineScreenState = selectTrainLineViewModel::updateTrainLineScreenState,
-            showAlertDialog = selectTrainLineViewModel::showAlertDialog,
-            modifier = Modifier.padding(it)
+            onTrainLineSelected = onTrainLineSelected,
```
great coding practice, very nice clean-up and great changes according to previous discussion!

lilianlyx commented on app/src/main/java/com/example/ptvproject/ui/selecttrainline/SelectTrainLineScreen.kt:
```diff
@@ -64,8 +55,8 @@ fun SelectTrainLineContent(
     updateTrainLineScreenState: () -> String,
-    showAlertDialog: () -> Unit,
-) {
+    onTrainLineSelected: (String, String, String, String) -> Unit,
```
same comment as [above](https://github.com/LotusGrads/IamLateForTheTrain/pull/26/files?diff=split&w=1#r1220918162)

lilianlyx commented on app/src/main/java/com/example/ptvproject/ui/selecttrainline/SelectTrainLineScreen.kt:
```diff
@@ -140,8 +133,9 @@ fun SelectTrainLineContent(
-    showAlertDialog: () -> Unit,
     modifier: Modifier = Modifier,
+    trainLineUiState: TrainLineUiState,
+    onTrainLineSelected: (departureTime: String, stationName: String, trainLine: String, direction: String) -> Unit,
```
beautifully written lambda!

lilianlyx commented on app/src/main/java/com/example/ptvproject/ui/theme/Type.kt:
thanks for adding and fixing up the typography! can't wait to see it in action

lilianlyx reviewed (approved):
discusses changes were actioned really well, just some minor comments on using parameter names for the provided types (as commented in the code). great work!

lilianlyx reviewed (approved):
discusses changes were actioned really well, just some minor comments on using parameter names for the provided types (as commented in the code). great work!

Commit:
Changes to PTV App file so that stopName is properly parsed.
Changes to SelectTrainLineScreen/UI State/ViewModel to ensure that there are 3 states (including loading state).
Converting time to readable format.

lilianlyx reviewed (approved):
clear changes, some commenting could help clarify additions to the code, but otherwise well-written and a purposeful new commit
